lib/spack/spack/installer.py

```diff
@@ -1366,10 +1365,6 @@ def _prepare_for_install(self, task: BuildTask) -> None:
                 else:
                     tty.debug(f"{task.pkg_id} is partially installed")
 
-            # Destroy the stage for a locally installed, non-DIYStage, package
-            if restage and task.pkg.stage.managed_by_spack:
```

Comment:
This is moved into `BuildProcessInstaller.run()` so we don't dereference `task.pkg.stage` when doing binary installs